### ISSUE TITLE
Elements: How they should be used

### ISSUE THREAD

**Issue by mhartz** [tmpworldwide/tmpworldwide.github.io#14]
Here is the proposed additions to the Code Standards:

3.2  Elements
3.2.1  `<nav>`
`<nav>` tags are designed to indicate major navigation blocks on a webpage, such as a menu bar or pagination. They shouldn't be applied to regular links, only to significant sections that are wholly dedicated to navigation.

3.2.2  `<header>`
`<header>` elements typically contain the section’s heading (`<h1>` - `<h6>` or an `<hgroup>` element) but can also contain other content such as a table of contents, a search form or relevant logos. You are not restricted to using one <header> element per site. You can use multiple <header> tags which will then become the header for the section of that document. ie:

```
<header>
  <h1>Most important heading on page</h1>
</header>
<article>
  <header>
    <h1>Title of Article</h1>
  </header>
  <p>Article text</p>
</article>
```

3.2.3  `<main>`
Must not have more than one `<main>` element on a page.
Must not be a child of an `<article>`, `<aside>`, `<footer>`, `<header>`, or `<nav>` element.

3.2.4  `<aside>`
`<aside>` tags could effectively be described as the "everything else" element for content that is neither article, section nor navigation. W3C prescribes this tag for use when something is tangentially related to the content around it, but is separate from the main content of the page, such as sidebars, blogrolls and so on.

3.2.5  `<article>`
`<article>` tags should be used where a piece of content could be taken out of a page completely and still make sense by itself with no surrounding content.

3.2.4  `<section>`
`<section>` tags should be used for content that is grouped together according to a theme, but makes sense only in the context of the content surrounding it. Element’s contents would be listed explicitly in the document’s outline. All <section> tags must contain at least one heading (`<h1>` - `<h6>`) element.

3.2.6  `<div>`
The `<div>` element is a generic container for flow content that by itself does not represent anything. It is best used for applying css styles, as layout containers, or as divisions to make HTML easier to read. It can be considered the element of last resort, for when no other element is suitable due to accessibility and poor maintainability.

**Comment by michaelspellacy** [comment]
I believe hgroup has been removed from specification. Also, until we get solid data telling us otherwise, we should include landmark aria roles on certain elements. The primary header of a document might contain a role of "banner". The main element should contain a role of "main", etc. While some may seem redundant, we define these landmark areas for assistive technology  that may not yet understand HTML5 elements. Trying to do some research on what elements are supported by AT now. Will keep you posted.

**Comment by mhartz** [comment]
I'm thinking things such as roles would be in another section. I'm thinking this section isn't really a guide on what to do, but what your limitations are when using these elements.

I think that roles should have a section itself.

**Comment by scampbell44** [comment]
Looks good and I agree that roles and other such attributes for the elements should be defined elsewhere. I like the structural succinctness of this document.

**Comment by michaelspellacy** [comment]
That's fine, for most other roles I would agree, however, I think it should it be noted in this section, towards the end. For benefit of those developers who may not be aware of their importance in landmarks like these.

**Comment by scampbell44** [comment]
Perhaps a brief mention with a link to the fuller presentation.

**Comment by michaelspellacy** [comment]
Yeah, for sure. I try to be very mindful of the code I put out into the world as it can proliferate quickly. This is why i care about calling it out. I don't want our code to be blamed for creating a less accessible site elsewhere. :-)

**Comment by mhartz** [comment]
Should we add this in, and add in the roles as a new issue, but in the same section for now?
